src/ast/mod.rs

```diff
@@ -566,15 +607,100 @@ impl fmt::Display for Statement {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             Statement::Query(s) => write!(f, "{}", s),
+            Statement::Msck {
+                table_name,
+                repair,
+                add_partitions,
+                drop_partitions,
+                sync_partitions,
+            } => {
+                write!(
+                    f,
+                    "MSCK {repair}TABLE {table}",
+                    repair = if *repair { "REPAIR " } else { "" },
+                    table = table_name
+                )?;
+                write!(
+                    f,
+                    "{add}{drop}{sync}",
+                    add = if *add_partitions {
+                        " ADD PARTITIONS"
+                    } else {
+                        ""
+                    },
+                    drop = if *drop_partitions {
+                        " DROP PARTITIONS"
+                    } else {
+                        ""
+                    },
+                    sync = if *sync_partitions {
+                        " SYNC PARTITIONS"
+                    } else {
+                        ""
+                    }
+                )
+            }
+            Statement::Truncate {
+                table_name,
+                partitions,
+            } => {
+                write!(f, "TRUNCATE TABLE {}", table_name)?;
+                if let Some(ref parts) = partitions {
+                    if !parts.is_empty() {
```

Comment:
Is `PARTITIONS ()` also possible? If so, then this check is not needed. Otherwise, we could check the partitions type to a vec instead of optional vec.

Reply:
I don't believe `PARTITIONS ()` is valid, you have to specify at the very least a column.